frontend/src/v5/ui/routes/dashboard/projects/federations/federationSettingsForm/federationSettingsForm.component.tsx

```diff
@@ -0,0 +1,405 @@
+					}))
+	)),
+	unit: Yup.string().required().default('mm'),
+	code: Yup.lazy((value) => (
+		value === ''
+			? Yup.string().strip()
+			: Yup.string()
+				.min(1,
+					formatMessage({
+						id: 'federations.settings.code.error.min',
+						defaultMessage: 'Code must be at least 1 character',
+					}))
+				.max(50,
+					formatMessage({
+						id: 'federations.settings.code.error.max',
+						defaultMessage: 'Code is limited to 50 characters',
+					}))
+				.matches(/^[\w|_|-]*$/,
+					formatMessage({
+						id: 'federations.settings.code.error.characters',
+						defaultMessage: 'Code can only consist of letters and numbers',
+					}))
+	)),
+	defaultView: Yup.string()
+		.nullable()
+		.transform((value) => (value === EMPTY_VIEW._id ? null : value)),
+	latitude: Yup.number().required(),
+	longitude: Yup.number().required(),
+	angleFromNorth: Yup.number()
+		.min(0,
+			formatMessage({
+				id: 'federations.settings.angle.error.min',
+				defaultMessage: 'Angle cannot be smaller than 0',
+			}))
+		.max(360,
+			formatMessage({
+				id: 'federations.settings.angle.error.max',
+				defaultMessage: 'Angle cannot be greater than 360',
+			}))
+		.transform((value) => value ?? 0),
+	x: Yup.number().required(),
+	y: Yup.number().required(),
+	z: Yup.number().required(),
+});
+
+const getThumbnailBasicPath = (teamspace: string, projectId: string, federationId: string) => (
+	(viewId: string) => `teamspaces/${teamspace}/projects/${projectId}/federations/${federationId}/views/${viewId}/thumbnail`
+);
+
+type IFederationSettingsForm = {
+	open: boolean;
+	federation: IFederation;
+	onClose: () => void;
+};
+
+export const FederationSettingsForm = ({ open, federation, onClose }: IFederationSettingsForm) => {
+	let defaultValues = getDefaultValues(federation) as any;
+	const {
+		register,
+		handleSubmit,
+		reset,
+		watch,
+		control,
+		formState,
+		formState: { errors },
+	} = useForm<IFormInput>({
+		mode: 'onChange',
+		resolver: yupResolver(FederationSchema),
+		defaultValues,
+	});
+
+	const currentUnit = watch('unit');
+
+	const { teamspace, project } = useParams() as { teamspace: string, project: string };
+	const getThumbnail = getThumbnailBasicPath(teamspace, project, federation._id);
+
+	useEffect(() => {
+		defaultValues = getDefaultValues(federation) as any;
+		reset(defaultValues);
+		FederationsActionsDispatchers.fetchFederationSettings(teamspace, project, federation._id);
+		FederationsActionsDispatchers.fetchFederationViews(teamspace, project, federation._id);
+	}, [open]);
+
+	const onSubmit: SubmitHandler<IFormInput> = ({
+		latitude, longitude,
+		x, y, z,
+		...otherSettings
+	}) => {
+		const payload: RawFederationSettings = {
+			surveyPoints: [{
+				latLong: [latitude, longitude],
+				position: [x, y, z],
+			}],
+			...otherSettings,
+		};
+		FederationsActionsDispatchers.updateFederationSettings(teamspace, project, federation._id, payload);
```

Comment:
lets move this data transformation to updateFederationSettings in sagas.

Reply:
done